varnish-macros/src/gen_func.rs

```diff
@@ -372,12 +372,23 @@ impl FuncProcessor {
             // This ensures the string is properly escaped and surrounded by quotes
             default = default.to_string().into();
         }
-        let mut json_arg: Vec<Value> = vec![
-            vcc_type.into(),
-            arg_name.into(),
-            default,
-            Value::Null, // spec param is not used at this point
-        ];
+
```

Comment:
I wonder if this would work?

```rust
let mut json_arg: Vec<Value> = vec![
            vcc_type.into(),
            #[cfg(varnishsys_vmod_meta_1_0)]
            arg_name.clone().into(),
            arg_name.into(),
            default,
            Value::Null, // spec param is not used at this point
        ];
```

Reply:
@nyurik - That works, though the logic needs to be reversed. I've updated the pull.